File: src/main/java/icu/samnyan/aqua/sega/chusan/handler/GetGameMapAreaConditionHandler.java
package icu.samnyan.aqua.sega.chusan.handler.impl;

import com.fasterxml.jackson.core.JsonProcessingException;
import icu.samnyan.aqua.sega.chusan.handler.BaseHandler;
import icu.samnyan.aqua.sega.util.jackson.StringMapper;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import java.util.LinkedHashMap;
import java.util.Map;

@Component("ChusanGetGameMapAreaConditionHandler")
public class GetGameMapAreaConditionHandler implements BaseHandler {
    private static final Logger logger = LoggerFactory.getLogger(GetGameMapAreaConditionHandler.class);

    private final StringMapper mapper;

    @Autowired
    public GetGameMapAreaConditionHandler(StringMapper mapper) {
        this.mapper = mapper;
    }

    @Override
    public String handle(Map<String, Object> request) throws JsonProcessingException {
        logger.info("MapAreaCondition Dummy Handler");

        Map<String, Object> resultMap = new LinkedHashMap<>();
        resultMap.put("mapAreaConditionList", new LinkedHashMap<>());

        String json = mapper.write(resultMap);
        logger.info("Response: " + json);
        return json;
    }
}

File: src/main/java/icu/samnyan/aqua/sega/chusan/handler/GetGameMapAreaConditionHandler.kt
package icu.samnyan.aqua.sega.chusan.handler

import ext.logger
import icu.samnyan.aqua.sega.general.BaseHandler
import icu.samnyan.aqua.sega.util.jackson.StringMapper
import org.springframework.stereotype.Component

@Component("ChusanGetGameMapAreaConditionHandler")
class GetGameMapAreaConditionHandler(val mapper: StringMapper) : BaseHandler {
    override fun handle(request: Map<String, Any>): String {
        val cond = listOf(
            mapOf("mapAreaId" to 2206201, "mapAreaConditionList" to listOf(
                mapOf("type" to 3, "conditionId" to 6832, "logicalOpe" to 1)
            )),
            mapOf("mapAreaId" to 2206203, "mapAreaConditionList" to listOf(
                mapOf("type" to 3, "conditionId" to 6833, "logicalOpe" to 1)
            )),
            mapOf("mapAreaId" to 2206204, "mapAreaConditionList" to listOf(
                mapOf("type" to 3, "conditionId" to 6834, "logicalOpe" to 1),
                mapOf("type" to 3, "conditionId" to 6835, "logicalOpe" to 1)
            )),
            mapOf("mapAreaId" to 2206205, "mapAreaConditionList" to listOf(
                mapOf("type" to 3, "conditionId" to 6837, "logicalOpe" to 1)
            )),
            mapOf("mapAreaId" to 2206206, "mapAreaConditionList" to listOf(
                mapOf("type" to 3, "conditionId" to 6838, "logicalOpe" to 1)
            )),
            mapOf("mapAreaId" to 2206207, "mapAreaConditionList" to listOf(
                mapOf("type" to 2, "conditionId" to 2206201, "logicalOpe" to 1),
                mapOf("type" to 2, "conditionId" to 2206202, "logicalOpe" to 1),
                mapOf("type" to 2, "conditionId" to 2206203, "logicalOpe" to 1),
                mapOf("type" to 2, "conditionId" to 2206204, "logicalOpe" to 1),
                mapOf("type" to 2, "conditionId" to 2206205, "logicalOpe" to 1),
                mapOf("type" to 2, "conditionId" to 2206206, "logicalOpe" to 1)
            )),
            mapOf("mapAreaId" to 3229301, "mapAreaConditionList" to listOf(
                mapOf("type" to 1, "conditionId" to 3020701, "logicalOpe" to 2)
            )),
            mapOf("mapAreaId" to 3229302, "mapAreaConditionList" to listOf(
                mapOf("type" to 1, "conditionId" to 3020701, "logicalOpe" to 1)
            ))
        )

        val resultMap: MutableMap<String, Any> = linkedMapOf(
            "gameMapAreaConditionList" to cond
        )

        val json = mapper.write(resultMap)
        logger.info("Response: $json")
        return json
    }

    companion object {
        private val logger = logger()
    }
}
